### PR TITLE
consensus: added byzantine test, modified previous one

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

Add a simple double vote scenario where a validator sends a nil vote to one half of the validators and a vote for the proposed block to the other half of the validators - then checks that evidence is committed in the block thereafter

Also modified the existing byzantine test (which has a partition where a byzantine node submits different proposals) to
be a little more descriptive and to ensure that the byzantine node actually submits different proposals

### CODE REVIEW

File: consensus/byzantine_test.go [hunk 1]
```diff
@@ -3,32 +3,152 @@ package consensus
...
+	"github.com/tendermint/tendermint/store"
 	"github.com/tendermint/tendermint/types"
 )
 
 //----------------------------------------------
 // byzantine failures
 
+// Byzantine node sends two different prevotes (nil and blockID) to the same validator
+func TestByzantinePrevoteEquivocation(t *testing.T) {
+	nValidators := 4
```
Comment (melekes) [inline consensus/byzantine_test.go:36]:
```suggestion
	const nValidators = 4
```

File: consensus/byzantine_test.go [hunk 2]
```diff
@@ -3,32 +3,152 @@ package consensus
...
+	defer stopConsensusNet(log.TestingLogger(), reactors, eventBuses)
+
+	// create byzantine validator
+	bcs := css[0]
+
+	bcs.doPrevote = func(height int64, round int32) {
+		// allow first height to happen normally so that byzantine validator is no longer proposer
+		if height == 2 {
+			bcs.Logger.Info("Sending two votes")
+			// bcs.mtx.Lock()
```
Comment (melekes) [inline consensus/byzantine_test.go:105]: remove?

File: consensus/byzantine_test.go [hunk 3]
```diff
@@ -3,32 +3,152 @@ package consensus
...
+			prevote1, err := bcs.signVote(tmproto.PrevoteType, bcs.ProposalBlock.Hash(), bcs.ProposalBlockParts.Header())
+			require.NoError(t, err)
+			prevote2, err := bcs.signVote(tmproto.PrevoteType, nil, types.PartSetHeader{})
+			require.NoError(t, err)
+			// bcs.mtx.Unlock()
+			peerList := reactors[1].Switch.Peers().List()
+			bcs.Logger.Info("Getting peer list", "peers", peerList)
+			// send two votes to all peers
```
Comment (melekes) [inline consensus/byzantine_test.go:113]:
```suggestion
			// send two votes to all peers (1st to one half, 2nd to another half)
```

File: consensus/byzantine_test.go [hunk 4]
```diff
@@ -3,32 +3,152 @@ package consensus
...
+	// Check that evidence is submitted and committed at the third height
+	for i := 0; i < 2; i++ {
+		timeoutWaitGroup(t, nValidators, func(j int) {
+			<-blocksSubs[j].Out()
+		}, css)
+	}
+	timeoutWaitGroup(t, nValidators, func(j int) {
+		msg := <-blocksSubs[j].Out()
+		block := msg.Data().(types.EventDataNewBlock).Block
+		assert.True(t, len(block.Evidence.Evidence) == 1)
```
Comment (melekes) [inline consensus/byzantine_test.go:138]: can we check the evidence type?

File: consensus/byzantine_test.go [hunk 5]
```diff
@@ -3,32 +3,152 @@ package consensus
...
+	bcs.doPrevote = func(height int64, round int32) {
+		// allow first height to happen normally so that byzantine validator is no longer proposer
+		if height == 2 {
+			bcs.Logger.Info("Sending two votes")
+			// bcs.mtx.Lock()
+			prevote1, err := bcs.signVote(tmproto.PrevoteType, bcs.ProposalBlock.Hash(), bcs.ProposalBlockParts.Header())
+			require.NoError(t, err)
+			prevote2, err := bcs.signVote(tmproto.PrevoteType, nil, types.PartSetHeader{})
+			require.NoError(t, err)
+			// bcs.mtx.Unlock()
```
Comment (melekes) [inline consensus/byzantine_test.go:110]: remove?

File: consensus/byzantine_test.go (hunk 4 above)
Comment (cmwaters) [reply consensus/byzantine_test.go:139]: I also now check that the addresses match

**Pull request merged**